### PR TITLE
fix: Show an error message when incompatible winappsdk to net7.0 with uno is found

### PR DESCRIPTION
**Pull request by jeromelaban**
## PR Type

What kind of change does this PR introduce?
- Bugfix

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: build/uno.winui.winappsdk.targets [hunk 1]
```diff
@@ -64,4 +64,28 @@
 
 	</Target>
 
+	<Target Name="_FindInvalidWinAppSDKUnoPlatformReference"
+			BeforeTargets="ResolveAssemblyReferences"
+			Condition="'$(UnoDisableUNOB0002Validation)'==''">
```
Comment (Youssef1313) [inline build/uno.winui.winappsdk.targets:69]:
```suggestion
			Condition="'$(UnoDisableUNOB0002Validation)'!='true'">
```

File: doc/articles/uno-build-error-codes.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
+uid: Build.Solution.error-codes
+---
+# Uno Build error codes
+
+## UNOB0001: Cannot build with both Uno.WinUI and Uno.UI NuGet packages referenced
+
+This error code means that a project has determined what both `Uno.WinUI` and `Uno.UI` packages are referenced.
+
+To fix this issue, you may be explicitly referencing `Uno.UI` and `Uno.WinUI` in your `csproj`, or you may be referencing a NuGet package which is incompatible with your current project's configuration.
```
Comment (agneszitte-nventive) [inline doc/articles/uno-build-error-codes.md:10]:
```suggestion
To fix this issue, you may be explicitly referencing `Uno.UI` and `Uno.WinUI` in your `csproj`, or you may be referencing a NuGet package that is incompatible with your current project's configuration.
```

File: doc/articles/uno-build-error-codes.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+## UNOB0001: Cannot build with both Uno.WinUI and Uno.UI NuGet packages referenced
+
+This error code means that a project has determined what both `Uno.WinUI` and `Uno.UI` packages are referenced.
+
+To fix this issue, you may be explicitly referencing `Uno.UI` and `Uno.WinUI` in your `csproj`, or you may be referencing a NuGet package that is incompatible with your current project's configuration.
+
+For instance, if your project references `Uno.WinUI`, and you try to reference `SkiaSharp.View.Uno`, you will get this error. To fix it, you'll need to reference `SkiaSharp.View.Uno.WinUI` instead.
+
+## UNOB0002: Project XX contains a reference to Uno Platform but does not contain a WinAppSDK compatible target framework.
```
Comment (Xiaoy312) [inline doc/articles/uno-build-error-codes.md:14]: does https://aka.platform.uno/UNOB0002 redirect here somehow?

**Review by Xiaoy312 — APPROVED** [review]

File: doc/articles/uno-build-error-codes.md (hunk 3 above)
Comment (jeromelaban) [reply doc/articles/uno-build-error-codes.md:14]: It will be redirecting to that exact spot, once the docs are published :)

**Pull request merged**